lib/dangermattic/plugins/tracks_checker.rb

```diff
@@ -60,10 +60,24 @@ def changes_tracks_files?(tracks_files:)
     end
 
     def diff_has_tracks_changes?(tracks_usage_matchers:)
+      tracks_changes_on_additions = diff_has_tracks_changes_for_change_type?(
+        tracks_usage_matchers: tracks_usage_matchers,
+        change_type: :added
+      )
+
+      tracks_changes_on_removals = diff_has_tracks_changes_for_change_type?(
+        tracks_usage_matchers: tracks_usage_matchers,
+        change_type: :removed
+      )
+
+      tracks_changes_on_additions || tracks_changes_on_removals
+    end
+
+    def diff_has_tracks_changes_for_change_type?(tracks_usage_matchers:, change_type:)
       matched_lines = git_utils.matching_lines_in_diff_files(
         files: git_utils.all_changed_files,
         line_matcher: ->(line) { tracks_usage_matchers.any? { |tracks_usage_match| line.match(tracks_usage_match) } },
-        change_type: nil
+        change_type: change_type
```

Comment:
I wonder if it wouldn't be worth it to update `git_utils.matching_lines_in_diff_files` so that its `change_type` parameter could be either a single symbol (like it currently is)… or an array of symbols?

That way you'd be able to revert your changes above in this `tracks_checker.rb` code (having to create a DRY function and call it twice with 2 different `change_type`) and instead would change the trunk code from `change_type: nil` to `change_type: [:added, :removed]`.

Maybe I'm thinking too much ahead, but I feel like asking for both additions and deletions could be useful in contexts and plugins/checks others than the one for Tracks.

Besides, the change to make `matching_lines_in_diff_files` accept an array would likely be simpler than the changes you did above after all 😛 :
 - Add `change_types = Array(change_type).map(&:to_sym)` at the start of the method, so that if `change_type` is a single value (String or Symbol) it will be wrapped inside an `Array`; while if it's already an `Array` it won't do anything to it. (and if it's `nil` that will produce an empty Array `[]`)
 - Change your condition in `matched_lines << line if …` from including `change_type(diff_line: line) == change_type` to `change_types.include?(change_type(diff_line: line))`

Reply:
It actually crossed my mind, but that would break the API -- though I'm 100% certain no one uses this method, so I wouldn't do a 2.0 just for this anyway 😅 . 

Another alternative would be to add a different symbol, like `:added_or_removed`, which I think would cover most cases (it's hard to think of someone filtering for "added or context"), but still it wouldn't be as flexible as an array.

Reply:
Would that break the API though ? The solution I proposed would still accept **either** an Array **or** a single item, so I think that's ok?

I mean, that's the whole reason why I specifically suggested that implementation with `Array(x)`, as using `change_types = Array(change_type)` will guarantee backward-compatibility. You would still be able to pass a single value like `:added` at call site (this keeping backwards-compat), and `change_types = Array(:added)` will turn wrap the input parameter and end up with `change_types = [:added]` for the rest of your implementation to use it as an array of one element; but now you _also_ have to possibility for your caller to pass an array like `[:added, :removed]` and `change_types = Array([:added, :removed]` will set `change_types = [:added, :removed]` without wrapping the parameter when it's already an Array.

Reply:
Ahh, gotcha, I haven't realized it wouldn't break the API. Thanks for the suggestion, I'll do it! 👍

Reply:
Updated on feeb9f9e17089871237bd86182520435384eed9f